lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -220,6 +222,9 @@ defmodule ExDoc.Formatter.HTML.Autolink do
   """
   def project_doc(bin, modules, module_id \\ nil) when is_binary(bin) do
     project_funs = for m <- modules, d <- m.docs, do: doc_prefix(d) <> m.id <> "." <> d.id
+    project_types = for m <- modules, d <- m.typespecs, do: "t:" <> m.id <> "." <> d.id
```

Comment:
Callbacks go into `docs`, otherwise linking to callbacks wouldn't work (as they did earlier). Specs are part of a function node, you can retrieve it via `d.specs` in the above code snippet. So, `typespecs` only contain the types defined in the module.